From b7a2e356930871310bd37c14f66dd47a963d6847 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cs=C3=B3k=C3=A1s=2C=20Bence?= <csokas.bence@prolan.hu>
Date: Mon, 9 Sep 2024 12:32:03 +0200
Subject: [PATCH] udev-rules: Fix compile errors with `-DDEBUG=1`
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

`DEBUG` turns on some - apparently untested - code that
causes compile errors (see linked GitHub issue). Fixing up
the unfinished migration to `strbuf`s solves this problem.

Fixes: 915bf0f60fa7 ("udev: use strbuf to store rules strings")
Closes: #287
Signed-off-by: Csókás, Bence <csokas.bence@prolan.hu>
---
 src/udev/udev-rules.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/udev/udev-rules.c b/src/udev/udev-rules.c
index 3856686c8..c0af42c98 100644
--- a/src/udev/udev-rules.c
+++ b/src/udev/udev-rules.c
@@ -326,8 +326,8 @@ static void dump_token(struct udev_rules *rules, struct token *token) {
         enum token_type type = token->type;
         enum operation_type op = token->key.op;
         enum string_glob_type glob = token->key.glob;
-        const char *value = str(rules, token->key.value_off);
-        const char *attr = &rules->buf[token->key.attr_off];
+        const char *value = rules_str(rules, token->key.value_off);
+        const char *attr = rules_str(rules, token->key.attr_off);
 
         switch (type) {
         case TK_RULE:
@@ -337,9 +337,9 @@ static void dump_token(struct udev_rules *rules, struct token *token) {
                         unsigned int idx = (tk_ptr - tks_ptr) / sizeof(struct token);
 
                         log_debug("* RULE %s:%u, token: %u, count: %u, label: '%s'",
-                                  &rules->buf[token->rule.filename_off], token->rule.filename_line,
+                                  rules_str(rules, token->rule.filename_off), token->rule.filename_line,
                                   idx, token->rule.token_count,
-                                  &rules->buf[token->rule.label_off]);
+                                  rules_str(rules, token->rule.label_off));
                         break;
                 }
         case TK_M_ACTION:
@@ -436,11 +436,11 @@ static void dump_token(struct udev_rules *rules, struct token *token) {
 static void dump_rules(struct udev_rules *rules) {
         unsigned int i;
 
-        log_debug("dumping %u (%zu bytes) tokens, %u (%zu bytes) strings",
+        log_debug("dumping %u tokens (%zu bytes), %zu strings (%zu bytes)",
                   rules->token_cur,
                   rules->token_cur * sizeof(struct token),
-                  rules->buf_count,
-                  rules->buf_cur);
+                  rules->strbuf->in_count,
+                  rules->strbuf->in_len);
         for (i = 0; i < rules->token_cur; i++)
                 dump_token(rules, &rules->tokens[i]);
 }
